src/txn/txn.c

```diff
@@ -126,7 +126,8 @@ __wt_txn_release_snapshot(WT_SESSION_IMPL *session)
 
     WT_ASSERT(session, txn_shared->pinned_id == WT_TXN_NONE ||
         session->txn->isolation == WT_ISO_READ_UNCOMMITTED ||
-        !__wt_txn_visible_all(session, txn_shared->pinned_id, WT_TS_NONE));
+        !__wt_txn_visible_all(session, txn_shared->pinned_id, WT_TS_NONE) ||
+        F_ISSET(session, WT_SESSION_HS_CURSOR));
```

Comment:
If we have a history store cursor, we may have pinned ids in a situation where we might not have before. Therefore, this need to be added as a possible case in the assertion.